File: backend/src/domain/entities/assignment.py
import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class AssignmentStatus(str, Enum):
    DRAFT = "DRAFT"
    PENDING_REVIEW = "PENDING_REVIEW"
    IN_REVIEW = "IN_REVIEW"
    REQUIRES_CHANGES = "REQUIRES_CHANGES"
    APPROVED = "APPROVED"


class AssignmentEntity(BaseModel):
    id: uuid.UUID
    student_id: uuid.UUID
    teacher_id: uuid.UUID
    title: str
    status: AssignmentStatus
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AssignmentVersionEntity(BaseModel):
    id: uuid.UUID
    assignment_id: uuid.UUID
    version_number: int
    content: dict
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/domain/models/assignment.ts
export type AssignmentStatus =
  | 'DRAFT'
  | 'PENDING_REVIEW'
  | 'IN_REVIEW'
  | 'REQUIRES_CHANGES'
  | 'APPROVED'

export interface Assignment {
  id: string
  student_id: string
  teacher_id: string
  title: string
  status: AssignmentStatus
  created_at: string
  updated_at: string
}

export interface AssignmentVersion {
  id: string
  assignment_id: string
  version_number: number
  content: TipTapNode
  created_at: string
}

export interface TipTapNode {
  type: string
  content?: TipTapNode[]
  text?: string
  attrs?: Record<string, unknown>
  marks?: Array<{ type: string; attrs?: Record<string, unknown> }>
}
